File: C++/Tickets to Ride/tickettoride-framework-r6/java/client/eclipse/src/datastruct/TTRNode.java

package datastruct;

/**
 * Node class.
 * Represents towns and stations.
 */
public class TTRNode {
	
	/**
	 * Node id.
	 */
	private int  id = 0;
	
	/**
	 * Node name.
	 */
	private String name;
	
	/**
	 * Owning player (if any player built a station here).
	 */
	private	TTRPlayer p;
	
	/**
	 * Edge owned by this station (if any player built a station here).
	 */
	private	TTREdge e;
	
	/**
	 * Constructor.
	 * Sets name to parameter value and initializes members.
	 * @param name node name.
	 */
	public TTRNode(String name)
	{
		this.name = new String(name);
		p = null; e = null;
		id = -1;
	}
	
	/**
	 * Setter method for id member.
	 * @param node id.
	 */
	public void setId(int id)	
	{
		this.id = id;
	}
	
	/**
	 * Getter method for id member. 
	 * @return node id.
	 */
	public int getId()
	{
		return id;
	}
	
	/**
	 * Setter method for p member.
	 * @param p player reference.
	 */
	void setPlayer(TTRPlayer p) 
	{
		this.p = p;
	}
	
	/**
	 * Getter method for p member.
	 * @return p reference.
	 */
	TTRPlayer getPlayer()		
	{
		return p;
	}

	/**
	 * Setter method for e member.
	 * @param e edge reference.
	 */
	void setEdge(TTREdge e)		
	{
		this.e = e;
	}
	
	/**
	 * Getter method for e memeber.
	 * @return e reference.
	 */
	TTREdge getEdge()
	{
		return e;
	}

	/**
	 * Getter method for name member.
	 * @return node name.
	 */
	public String getName()
	{
		return name;
	}
}

File: C++/Tickets to Ride/tickettoride-framework-r6/c++/client/windows/project/TTRCppClient/TTRNode.cpp

#include "TTRNode.h"

TTRNode::TTRNode(char *_name){
	name = _name;
	p = (TTRPlayer *)0; e = (TTREdge *)0;
	id = -1;
}

void TTRNode::setId(int _id){
	id = _id;
}

int TTRNode::getId(void){
	return id;
}

void TTRNode::setPlayer(TTRPlayer *_p){
	p = _p;
}

TTRPlayer *TTRNode::getPlayer(void){
	return p;
}

void TTRNode::setEdge(TTREdge *_e){
	e = _e;
}

String TTRNode::getName(void){
	return name;
}

TTRNode::~TTRNode(void){

}